Formula/dotnet.rb

```diff
@@ -37,14 +36,24 @@ class Dotnet < Formula
 
   fails_with :gcc
 
+  # Fix build failure on ARM due to osx-x64 override and missing ILAsm/ILDAsm
+  patch :DATA
```

Comment:
Can we upstream this?

Reply:
Looks like we need a similar fix on Intel.
```
microsoft.netcore.app.2.1.0.csproj : error NU1101: Unable to find package runtime.osx-x64.microsoft.netcore.ildasm. No packages exist with this id in source(s): prebuilt, previously-source-built, reference-packages, source-built [/private/tmp/dotnet-sources/.dotnet/sdk/6.0.100/NuGet.targets]
```

Reply:
I think some of patch is actually a workaround for issues with upstream's bootstrap artifacts.

The common workaround also hit some issues as there wasn't even `RuntimeOsxX64*Version` defined, so I ended up using Linux's `RuntimeLinuxX64*Version` (rather than hardcode 6.0.0).

Need to discuss further with upstream on how these will be fixed.